cmd/geth/shardingcmd.go

```diff
@@ -1,46 +1,63 @@
 package main
 
 import (
-	"github.com/ethereum/go-ethereum/sharding/notary"
-	"github.com/ethereum/go-ethereum/sharding/proposer"
+	"fmt"
 
 	"github.com/ethereum/go-ethereum/cmd/utils"
+	"github.com/ethereum/go-ethereum/sharding/node"
+	"github.com/ethereum/go-ethereum/sharding/notary"
+	"github.com/ethereum/go-ethereum/sharding/observer"
+	"github.com/ethereum/go-ethereum/sharding/proposer"
 	cli "gopkg.in/urfave/cli.v1"
 )
 
 var (
-	notaryClientCommand = cli.Command{
-		Action:    utils.MigrateFlags(notaryClient),
-		Name:      "sharding-notary",
-		Aliases:   []string{"shard-notary"},
-		Usage:     "Start a sharding notary client",
+	shardingCommand = cli.Command{
+		Action:    utils.MigrateFlags(sharding),
+		Name:      "sharding",
+		Usage:     "Start a sharding-enabled node",
 		ArgsUsage: "[endpoint]",
-		Flags:     []cli.Flag{utils.DataDirFlag, utils.PasswordFileFlag, utils.NetworkIdFlag, utils.IPCPathFlag, utils.DepositFlag},
+		Flags:     []cli.Flag{utils.ActorFlag, utils.DataDirFlag, utils.PasswordFileFlag, utils.NetworkIdFlag, utils.IPCPathFlag, utils.DepositFlag},
 		Category:  "SHARDING COMMANDS",
 		Description: `
-Launches a sharding notary client that connects to a running geth node and submit collations to a Sharding Manager Contract. This feature is a work in progress.
-`,
-	}
-	proposerClientCommand = cli.Command{
-		Action:    utils.MigrateFlags(proposerClient),
-		Name:      "sharding-proposer",
-		Aliases:   []string{"shard-proposer"},
-		Usage:     "Start a sharding proposer client",
-		ArgsUsage: "[endpoint]",
-		Flags:     []cli.Flag{utils.DataDirFlag, utils.PasswordFileFlag, utils.NetworkIdFlag, utils.IPCPathFlag},
-		Category:  "SHARDING COMMANDS",
-		Description: `
-Launches a sharding proposer client that connects to a running geth node and proposes collations to notary node. This feature is a work in progress.
+Launches a sharding node that manages services related to submitting collations to a Sharding Manager Contract, notary and proposer services, and shardp2p connections. This feature is a work in progress.
 `,
 	}
 )
 
-func notaryClient(ctx *cli.Context) error {
-	c := notary.NewNotary(ctx)
-	return c.Start()
+// sharding is the main cmd line entry point for starting a sharding-enabled node.
+// A sharding node launches a suite of services including notary services,
+// proposer services, and a shardp2p protocol.
+func sharding(ctx *cli.Context) error {
+	// configures a sharding-enabled node using the cli's context.
+	shardingNode := node.NewNode(ctx)
+	registerShardingServices(shardingNode)
```

Comment:
`registerShardingServices` returns an error. Do we want to check for `if err != nil`?